webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/domain/logger/DebugComplianceLogger.scala

```diff
@@ -86,6 +87,48 @@ object ComplianceDebugLogger extends Logger {
 
   def node(id: NodeId): Logger = nodeCache.get(id.value)
 
+  implicit class RunAnalysisInfoToLog(val c: RunAnalysis) extends AnyVal {
+    import com.normation.rudder.domain.reports.RunAnalysisKind.*
+
+    def logId:  String = {
+      s"${c.expectedConfigId.fold("unknown id")(_.value)}/[${c.expectedConfigStart.fold("unknown start time")(
+          _.toIsoStringNoMillis
+        )}-${c.expirationDateTime.fold("unknown expiration date")(_.toIsoStringNoMillis)}]"
+    }
+    def logRun: String = {
+      s"${c.lastRunConfigId.fold("none or too old")(i => s"configId: ${i.value}")} received at: ${c.lastRunDateTime.fold("unknown")(
+          _.toIsoStringNoMillis
+        )}${c.lastRunExpiration.fold("")(r => s" | expired at ${r.toIsoStringNoMillis}")}"
+    }
```

Comment:
It's quite hard to see the pattern and the separating char with all the folds, maybe there should be local variables to makes things clearer ?